lib/utils.js

```diff
@@ -142,84 +141,77 @@ function getModule(filename, base) {
   return file;
 }
 
-function parseLines(lines, frame) {
-  frame.pre_context = lines.slice(Math.max(0, frame.lineno - (LINES_OF_CONTEXT + 1)), frame.lineno - 1);
-  frame.context_line = lines[frame.lineno - 1];
-  frame.post_context = lines.slice(frame.lineno, frame.lineno + LINES_OF_CONTEXT);
-}
+function readSourceFiles(filenames, cb) {
+  // we're relying on filenames being de-duped already
+  if (filenames.length === 0) return setTimeout(cb, 0, {});
 
-function readFilePromise(filename) {
-  return new Promise(function (resolve, reject) {
-    fs.readFile(filename, function (err, file) {
-      return err ? reject(err) : resolve(file);
+  var sourceFiles = {};
+  var numFilesToRead = filenames.length;
+  return filenames.forEach(function (filename) {
+    fs.readFile(filename, function (readErr, file) {
+      if (!readErr) sourceFiles[filename] = file.toString().split('\n');
```

Comment:
`if (!readErr)` guarantees that `file` is a [Buffer](https://nodejs.org/api/buffer.html#buffer_buf_tostring_encoding_start_end) according to [fs.readFile docs](https://nodejs.org/api/fs.html#fs_fs_readfile_file_options_callback), so I'm not concerned